src/menu/menu.c

```diff
@@ -161,10 +162,61 @@ static view_t *menu_get_view (menu_mode_t id) {
     return NULL;
 }
 
+static void draw_progress (float progress) {
+    surface_t *d = (progress >= 1.0f) ? display_get() : display_try_get();
+
+    if (d) {
+        rdpq_attach(d, NULL);
+
+        component_background_draw();
+
+        component_loader_draw(progress);
+
+        rdpq_detach_show();
+    }
+}
 
 void menu_run (boot_params_t *boot_params) {
     menu_init(boot_params);
 
+    // FIXME: rather than use a controller button, would it be better to use the cart button?
+    JOYPAD_PORT_FOREACH (port) { // FIXME: is this the best code location
```

Comment:
I completely don't understand why you need to poll the joypad 50 times. Once is enough and initial polling should be done in the `actions_init` function (if it's not already done).

Reply:
I dont like it, but was the only way I found to make it reliable... I will see what I can do. But To be fair, this could be a great 
opportunity to use the button instead...